website/src/theme/AnnouncementBar/index.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import React from 'react';
+import clsx from 'clsx';
+import {useThemeConfig, useAnnouncementBar} from '@docusaurus/theme-common';
+import {translate} from '@docusaurus/Translate';
+import IconClose from '@theme/IconClose';
+import styles from './styles.module.css';
+export default function AnnouncementBar() {
+  const {isActive, close} = useAnnouncementBar();
+  const {announcementBar, announcementBarActive, announcementBarLink} = useThemeConfig();
+
+  if (!isActive || !announcementBarActive) {
+    return null;
+  }
+
+  const {content, backgroundColor, textColor, isCloseable} = announcementBar;
+
+  return (
+    <div
+      className={styles.announcementBar}
+      style={{
+        backgroundColor,
+        color: textColor,
+      }}
+      role="banner">
+      {isCloseable && <div className={styles.announcementBarPlaceholder} />}
+      {announcementBarLink ? (
+        <a 
+          target="_blank" 
+          rel="noopener noreferrer" 
+          href={announcementBarLink}
+          className={styles.announcementBarLink}
+          style={{
+            backgroundColor,
+            color: textColor,
+          }}
+          >
+          <AnnouncementBarContent content={content} styles={styles} />
+        </a>
+      ) : (
+        <AnnouncementBarContent content={content} styles={styles} />
+      )}
+      
+      {isCloseable ? (
```

Comment:
I'm not sure I'm familiar with this syntax (the wrapped in a { } aspect, not the ternary operator -- I understand the latter) and I'd love to know more!

Reply:
The `{}` structure is the JSX way of using javavscript within html. So a common use case is for displaying html content conditionally based on a JS variable state. In this case, i'm adding additional html (the anchor link) if `announcementBarLink` is set in the config.

Here is a good read on when to use `{}` in JSX: https://reactjs.org/docs/introducing-jsx.html